ci-automation/release.sh

```diff
@@ -64,14 +74,94 @@ function _inside_mantle() {
 
     source ci-automation/ci_automation_common.sh
     source sdk_container/.repo/manifests/version.txt
+    CHANNEL="$(get_git_channel)"
+    VERSION="${FLATCAR_VERSION}"
+    azure_profile_config_file=""
+    secret_to_file azure_profile_config_file "${AZURE_PROFILE}"
+    azure_auth_config_file=""
+    secret_to_file azure_auth_config_file "${AZURE_AUTH_CREDENTIALS}"
+    aws_credentials_config_file=""
+    secret_to_file aws_credentials_config_file "${AWS_CREDENTIALS}"
+    aws_marketplace_credentials_file=""
+    secret_to_file aws_marketplace_credentials_file "${AWS_MARKETPLACE_CREDENTIALS}"
+    gcp_json_key_path=""
+    secret_to_file gcp_json_key_path "${GCP_JSON_KEY}"
+    google_release_credentials_file=""
+    secret_to_file google_release_credentials_file "${GOOGLE_RELEASE_CREDENTIALS}"
 
-    # TODO: set up credentials
-    # TODO: run mantle pre-release and release for all platforms
-    # (needs changes in mantle to consume from buildcache via https)
-    # TODO: run ore for AWS marketplace upload
+    for platform in aws azure; do
+      for arch in amd64 arm64; do
+        # Create a folder where plume stores flatcar_production_ami_*txt and flatcar_production_ami_*json
+        # for later push to bincache
+        rm -rf "${platform}-${arch}"
+        mkdir "${platform}-${arch}"
+        cd "${platform}-${arch}"
+        plume pre-release --force \
+          --debug \
+          --platform="${platform}" \
+          --aws-credentials="${aws_credentials_config_file}" \
+          --azure-profile="${azure_profile_config_file}" \
+          --azure-auth="${azure_auth_config_file}" \
+          --board="${arch}-usr" \
+          --channel="${CHANNEL}" \
+          --version="${FLATCAR_VERSION}" \
+          --write-image-list="images.json"
+        plume release \
+          --debug \
+          --aws-credentials="${aws_credentials_config_file}" \
+          --aws-marketplace-credentials="${aws_marketplace_credentials_file}" \
+          --publish-marketplace \
```

Comment:
we also need to find a way to map `product-id` with `arch` / `channel`. I was thinking about a JSON credential object, mapping both:
```json
{
  "alpha-arm64": "1234",
  "beta-amd64": "5678"
}
```

Then `--product-id $(cat products.json | jq ."${CHANNEL}-${arch}")` (same for the instance type, or this can be done programmatically from plume itself).

Reply:
Ok nice, the instance type is already programmatically defined: https://github.com/flatcar/mantle/blob/37a88ab2d960a38f31225a74f4bc72551d9b5c72/cmd/plume/release.go#L472-L476